### DISCUSSION TITLE
Authorization on a Column

### DISCUSSION THREAD

**Discussion by etmartinkazoo** [pocketbase/pocketbase#4373] [Q&A]
If you have a table, say `partners`, which has general information about a business' partners like `name`, `description`, `poc_email`, `poc_phone`, etc... how would you use authorization to hide sensitive information columns like `retainer_amount`, etc...

I have Pocketbase setup such that the `users` table is related to the `roles` table and then the API rules can be set to authorize only specific users to have access to a table based on their role and the filter rules. What I cannot seem to figure out is if I can authorize for specific columns too. Namely, only users with the role `admin` can read the column `retainer_amount`

I hope that makes sense. Great, great, great project, just trying to learn more. Thank you!

**Comment by ganigeorgiev** [comment]
There are no rules for individual fields.

If you want to keep some of the fields private you can move them to a separate collection that will have its own rules and a relation to the parent one.

Depending on your use case, an alternative could be also to "lock" the `users` collection and create a [View collection](https://pocketbase.io/docs/collections/#view-collection) (_it is read-only_) with subset of the original `users` fields.

_With the planned refactoring there eventually will be also a record marshalize/serialize hook to add/remove fields programmatically but there are no ETAs for this._

**Comment by etmartinkazoo** [comment]
@ganigeorgiev - Thank you for your feedback on this! I think I am going to keep it simple and just create a new collection `sensitive_partner_information` with a one to one relationship to `partners`.

Thank you again for your work on Pocketbase; it is a truly incredible project.

**Comment by etmartinkazoo** [comment]
@ganigeorgiev - Interestingly I cannot get authorization to work. I am using the following filter:
`@request.auth.roles ~ "pttf92lxdblylum"`

The user has an array of roles but I still cannot get it to work.
<img width="648" alt="image" src="https://github.com/pocketbase/pocketbase/assets/13721164/ab94dac4-6c77-4961-b714-e9d263c93dfb">
<img width="671" alt="image" src="https://github.com/pocketbase/pocketbase/assets/13721164/d31cf800-5814-4bb1-bfbd-e340ab4e998b">

**Reply by ganigeorgiev** [reply]
I don't know what "pttf92lxdblylum" is, what are the supported `roles` values, nor whether you are authenticated when sending the request.

Additionally, if you want an exact match with an id you can use the `?=` operator:

```js
@request.auth.roles.id ?= "pttf92lxdblylum"
```

**Reply by etmartinkazoo** [reply]
I apologize, I should have been more specific.
The `users` table has many `roles` and each `role` has many `users`. Effectively a many to many. In the example case, the logged in user has an associated role of id `pttf92lxdblylum` which is named 'Admin'.

I want to say, if the currently logged in user has the admin role - `pttf92lxdblylum` - then they should be authorized.

**Reply by ganigeorgiev** [reply]
I'm confused and no longer sure what actually is the problem you are trying to solve.

If you want to allow access to the "users" list and view actions only when the authenticated user has an "Admin" role, then using the below filter should suffice:

```js
// replace "name" with the name of your field that holds the "Admin" value
@request.auth.roles.name ?= "Admin"
```

If the above doesn't work, then either I'm misunderstanding what you are trying to do or you are not testing it correctly (eg. are you sure that you are authenticated when sending the request?).

If you need further help, please elaborate on the desired behavior and/or provide a code sample showing how you are sending the request.
